src/auth/src/user_provider/static_user_provider.rs

```diff
@@ -43,32 +35,8 @@ impl StaticUserProvider {
         })?;
         return match mode {
             "file" => {
```

Comment:
As we have a `mode` option for user provider, how about make `WatchFileUserProvider` a mode to reduce concepts of the top level? WDYT? @tisonkun @shuiyisong

Reply:
Good point, we can rename `StaticUserProvider` to something like `LocalUserProvider`(may be a breaking change) and change the original mode `file` to `watch_file`.

Reply:
I ever though of it. But I'd like to first be sure that "static file provider" and this "dynamic file provider" are different things and deserved expose to user for switching (that is, silently change static file provider to dynamic is unreasonable).

Then, it's about naming, and naming is hard :D

If we don't care compatibility, my proposal would be:

* `static_user_provider:foo=bar,faz=baz`
* `file_user_provider:static:file_path`
* `file_user_provider:dynamic:file_path`

But if we'd keep the compatibility, I prefer my current solution. Although I agree that we can break it now since I suppose it's not tightly depended.